docs/csharp/language-reference/compiler-messages/warning-waves.md

```diff
@@ -0,0 +1,165 @@
+
+## CS8826 - Partial method declarations have signature differences.
+
+*Warning level 6*
+
+This warning corrects some inconsistencies in reporting differences between partial method signatures. The compiler always reported an error when the partial method signatures created different CLR signatures. Now, the compiler reports CS8826 when the signatures are syntactically different C#.  Consider the following partial class:
+
+:::code language="csharp" source="./snippets/WarningWaves/WaveSix.cs" id="PartialMethodDeclaration":::
+
+The following partial class implementation generates several examples of CS8626:
+
+:::code language="csharp" source="./snippets/WarningWaves/WaveSix.cs" id="PartialMethodDefinition":::
+
+> [!NOTE]
+> If the implementation of a method uses a non-nullable reference type when the other declaration accepts nullable reference types, CS8611 is generated instead of CS8826.
+
+To fix any instance of these warnings, ensure the two signatures match.
+
+## CS7023 - A static type is used in an `is` or `as` expression.
+
+*Warning level 5*
+
+The `is` and `as` expressions always return `false` for a static type because you can't create instances of a static type. The following code produces CS7023:
+
+:::code language="csharp" source="./snippets/WarningWaves/WaveFive.cs" id="StaticTypeAsIs":::
+
+The compiler reports this warning because the type test can never succeed. To correct this warning, remove the test and remove any code executed only if the test succeeded. In the preceding example, the `else` clause is always executed. The method body could be replaced by that single line:
+
+```csharp
+Console.WriteLine("o is not an instance of a static class");
+```
+
+## CS8073 - The result of the expression is always 'false' (or `true`).
+
+*Warning level 5*
+
+The `==` and `!=` operators always return `false` (or `true`) when comparing an instance of a `struct` type to `null`. The following code demonstrates this warning. Assume `S` is a `struct` that has defined `operator ==` and `operator !=`:
+
+:::code language="csharp" source="./snippets/WarningWaves/WaveFive.cs" id="StructsArentNull":::
+
+To fix this error, remove the null check, and code that would execute if the object is `null`.
+
+## CS8848 - Operator 'from' can't be used here due to precedence. Use parentheses to disambiguate.
+
+*Warning level 5*
+
+The following examples demonstrate this warning. The expression binds incorrectly because of the precedence of the operators.
+
+:::code language="csharp" source="./snippets/WarningWaves/WaveFive.cs" id="QueryPrecedence":::
+
+To fix this error, put parentheses around the query expression:
+
+:::code language="csharp" source="./snippets/WarningWaves/WaveFive.cs" id="QueryPrecedenceNoWarn":::
+
+## Members must be fully assigned, use of unassigned variable (CS8880, CS8881, CS8882, CS8883, CS8884, CS8885, CS8886, CS8887)
```

Comment:
@tdykstra 

This section was very redundant if I had a separate header for each warning. In this commit, I tried to put some key words shared among the messages and added all the IDs. I'm ambivalent about that format. What do you think? Do you have any better suggestion here?

Reply:
This makes sense to me, though I'm ambivalent about it too, as it makes for a very long heading.  The only revision I might consider is to put the warning numbers first to match the sequence in the other headings. You could make it a range --

CS8880-8887

-- but I don't know if that would compromise SEO for 81-87

Reply:
Good point @tdykstra 

For now, I'll leave it as merged. I'll check stats and make any updates next month. I'll see how many searches we get on the warning code, vs. direct traffic from the F1 service.